connectors/src/connectors/slack/lib/retrieve_permissions.ts

```diff
@@ -0,0 +1,190 @@
+import type { Result } from "@dust-tt/client";
+import { Err, Ok } from "@dust-tt/client";
+
+import type { RetrievePermissionsErrorCode } from "@connectors/connectors/interface";
+import { ConnectorManagerError } from "@connectors/connectors/interface";
+import { getChannels } from "@connectors/connectors/slack/lib/channels";
+import { getSlackClient } from "@connectors/connectors/slack/lib/slack_client";
+import { slackChannelInternalIdFromSlackChannelId } from "@connectors/connectors/slack/lib/utils";
+import {
+  ExternalOAuthTokenError,
+  ProviderWorkflowError,
+} from "@connectors/lib/error";
+import { SlackChannel } from "@connectors/lib/models/slack";
+import logger from "@connectors/logger/logger";
+import { ConnectorResource } from "@connectors/resources/connector_resource";
+import { SlackConfigurationResource } from "@connectors/resources/slack_configuration_resource";
+import type { ConnectorPermission, ContentNode } from "@connectors/types";
+import { INTERNAL_MIME_TYPES } from "@connectors/types";
+
+export async function retrievePermissions({
+  connectorId,
+  parentInternalId,
+  filterPermission,
+}: {
+  connectorId: number;
+  parentInternalId: string | null;
+  filterPermission: ConnectorPermission | null;
+}): Promise<
+  Result<ContentNode[], ConnectorManagerError<RetrievePermissionsErrorCode>>
+> {
+  if (parentInternalId) {
+    return new Err(
+      new ConnectorManagerError(
+        "INVALID_PARENT_INTERNAL_ID",
+        "Slack connector does not support permission retrieval with non null `parentInternalId`"
+      )
+    );
+  }
+
+  const c = await ConnectorResource.fetchById(connectorId);
+  if (!c) {
+    logger.error({ connectorId }, "Connector not found");
+    return new Err(
+      new ConnectorManagerError("CONNECTOR_NOT_FOUND", "Connector not found")
+    );
+  }
+  const slackConfig =
+    await SlackConfigurationResource.fetchByConnectorId(connectorId);
+  if (!slackConfig) {
+    logger.error({ connectorId }, "Slack configuration not found");
+    // This is unexpected let's throw to return a 500.
+    throw new Error("Slack configuration not found");
+  }
+
+  let permissionToFilter: ConnectorPermission[] = [];
+
+  switch (filterPermission) {
+    case "read":
+      permissionToFilter = ["read", "read_write"];
+      break;
+    case "write":
+      permissionToFilter = ["write", "read_write"];
+      break;
+    case "read_write":
+      permissionToFilter = ["read_write"];
+      break;
+  }
+
+  const slackChannels: {
+    slackChannelId: string;
+    slackChannelName: string;
+    permission: ConnectorPermission;
+    private: boolean;
+  }[] = [];
+
+  try {
+    if (filterPermission === "read") {
+      const localChannels = await SlackChannel.findAll({
```

Comment:
Same remark as previous PR we want something much simpler for `slack_bot` a simple read of the required channels and no dependence to SlackChannel table.

Basically just "remoteChannels" aka simply `getChannels(slackClient, c.id, false)` more or less

Reply:
I'll see what I can do, I was under the impression that `SlackChannel` stores the state of which agent is the default but I might be mistaken